packages/format-library/src/text-color/index.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { select } from '@wordpress/data';
+import { BlockControls } from '@wordpress/editor';
+import {
+	applyFormat,
+	removeFormat,
+	getActiveFormat,
+} from '@wordpress/rich-text';
+import { Toolbar, IconButton, Popover, ColorPalette } from '@wordpress/components';
+
+const name = 'core/text-color';
+const title = __( 'Text Color' );
+
+export const textColor = {
+	name,
+	title,
+	tagName: 'span',
+	className: 'has-inline-color',
+	edit: class TextColorEdit extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.state = {
+				isOpen: false,
+			};
+		}
+
+		render() {
+			const {
+				value,
+				onChange,
+			} = this.props;
+
+			let activeColor;
+
+			const colors = get( select( 'core/block-editor' ).getSettings(), [ 'colors' ], [] );
+			const activeColorFormat = getActiveFormat( value, name );
+
+			if ( activeColorFormat ) {
+				const styleColor = activeColorFormat.attributes.style;
+				if ( styleColor ) {
+					activeColor = styleColor.replace( new RegExp( `^color:\\s*` ), '' );
+				}
+			}
+
+			return (
+				<>
+					<BlockControls>
```

Comment:
`BlockControls` shouldn't be used by a format. These are rich text level controls, not block level controls.

Reply:
Unfortunately, it's the only way to have the button display where it should be. Without wrapping the `Toolbar` with `BlockControls` it's displaying below the block. Let me know if you have better suggestions. Thanks!

Reply:
Sorry, I think I can do this. Forgot for a moment that I'm adding this on core and not separate extension. Thanks!